packages/frappe-ui-react/src/components/datePicker/useDatePicker.ts

```diff
@@ -1,26 +1,99 @@
 import { useCallback, useEffect, useMemo, useState } from "react";
-import { getDate, getDatesAfter, getDaysInMonth, getDateValue } from "./utils";
+import {
+  getDate,
+  getDatesAfter,
+  getDaysInMonth,
+  getDateValue,
+  getDateTimeValue,
+  formatDateTime12h,
+} from "./utils";
+import { DatePickerViewMode } from "./types";
+
+const MONTHS = [
+  "January",
+  "February",
+  "March",
+  "April",
+  "May",
+  "June",
+  "July",
+  "August",
+  "September",
+  "October",
+  "November",
+  "December",
+];
+
+function formatTime12h(hours: number, minutes: number): string {
+  const period = hours >= 12 ? "pm" : "am";
+  const h = hours % 12 || 12;
+  return `${h}:${minutes.toString().padStart(2, "0")} ${period}`;
+}
+
+function parseTimeValue(timeStr: string): { hours: number; minutes: number } {
+  if (!timeStr) return { hours: 0, minutes: 0 };
+  const [time, period] = timeStr.split(" ");
+  const [h, m] = time.split(":").map(Number);
+  let hours = h;
+  const periodLower = period?.toLowerCase();
+  if (periodLower === "pm" && h !== 12) hours += 12;
+  if (periodLower === "am" && h === 12) hours = 0;
+  return { hours, minutes: m };
+}
+
+function generateTimeOptions(): string[] {
+  const options: string[] = [];
+  for (let h = 0; h < 24; h++) {
+    for (let m = 0; m < 60; m += 15) {
+      options.push(formatTime12h(h, m));
+    }
+  }
+  return options;
+}
 
 export function useDatePicker({
```

Comment:
Please add documentation for this hook and any other function that is being exported